### PR TITLE
Fix README and tests

### PR DESCRIPTION
**Pull request by dancer-yuma**
# What I did

- add .gitignore
- revise some READMEs 
- fix tests if necessary

# Note

- I will fix python-02-ja only after this PR is merged.

### CODE REVIEW

File: 01_intermediate_functions_and_data_structures/01_challenge/test.py [hunk 1]
```diff
@@ -1,44 +1,37 @@
...
+from extract_data import extract_data, better_extract_data
+
 
 class TestExtractData(unittest.TestCase):
-    def test_extract_data_1(self):
+    def test_string(self):
         self.assertEqual(
-            extract_data("User details: Name:John. He is very active. He is also quite tall, standing at Height:181cm. He is Age:30, and spends his time as an Occupation:Engineer. End of data."),
-            ["Name:John", "Height:181cm", "Age:30", "Occupation:Engineer"]
+            extract_data(
```
Comment (dancer-yuma) [inline 01_intermediate_functions_and_data_structures/01_challenge/test.py:8]: `test_extract_data_1` and `test_extract_data_2` were testing with same kind of use cases, but each unit test should check the different use cases.

File: 01_intermediate_functions_and_data_structures/01_challenge/test.py [hunk 2]
```diff
@@ -1,44 +1,37 @@
...
-    def test_better_extract_data_3(self):
-        self.assertEqual(
-            better_extract_data("Emergency Contacts: Police:911, Fire:999, Medical:112. Stay safe."),
-            [("Police", "911"), ("Fire", "999"), ("Medical", "112")]
-        )
 
-if __name__ == '__main__':
-    unittest.main()
\ No newline at end of file
+if __name__ == "__main__":
```
Comment (dancer-yuma) [inline 01_intermediate_functions_and_data_structures/01_challenge/test.py:36]: Inconsistent formatting

File: 02_classes_modules_and_packages/01_challenge/README.md [hunk 3]
```diff
@@ -12,7 +12,7 @@ This class should allow adding books, removing books, and retrieving the current
 - **Interface**:
     - `add_book(title, author)`: Adds a new book.
     - `remove_book(title)`: Removes a book by title.
-    - `retrieve_books()`: Returns a list of all books in the library.
+    - `retrieve_books()`: Returns a list of of dictionaries that represent books in the library.
```
Comment (dancer-yuma) [inline 02_classes_modules_and_packages/01_challenge/README.md:15]: The original instruction was expecting a dictionary with the title as key and author as value, but that is not ideal data structure given that a book could have more information.

File: 02_classes_modules_and_packages/02_challenge/README.md [hunk 4]
```diff
@@ -37,12 +37,12 @@ inventory = Inventory()
 for vehicle in vehicles:
     inventory.add_vehicle(vehicle)
 
-inventory.apply_discount(lambda v: v.make == 'Toyota', 5)
+inventory.apply_discount(lambda v: v.make == "Toyota", 5)
 toyotas = inventory.search_vehicles("Corolla")
 
 for toyota in toyotas:
-    print(toyota)
+    toyota.display()
```
Comment (dancer-yuma) [inline 02_classes_modules_and_packages/02_challenge/README.md:44]: We have not covered `__str__` at this point, so it is better to implement some method to print car information.

File: 02_classes_modules_and_packages/02_challenge/README.md [hunk 5]
```diff
@@ -37,12 +37,12 @@ inventory = Inventory()
...
 
-for vehicle in inventory:
-    print(vehicle)
+for vehicle in inventory.retrieve_vehicles():
```
Comment (dancer-yuma) [inline 02_classes_modules_and_packages/02_challenge/README.md:46]: `inventory` is not a sequence.

File: 04_error_handling_and_debugging/02_challenge/event_scheduler.py [hunk 6]
```diff
@@ -3,10 +3,11 @@ def __init__(self, name, start_hour):
         self.name = name
         self.start_hour = start_hour
 
+
 def schedule_events(event_list):
     schedule = {}
     for event in event_list:
-        if event.start_hour < 0 or event.start_hour > 24:
+        if event.start_hour < 0 and event.start_hour > 24:
```
Comment (dancer-yuma) [inline 04_error_handling_and_debugging/02_challenge/event_scheduler.py:10]: I added one more twist, as there was only one bug in this code.

File: 04_error_handling_and_debugging/02_challenge/event_scheduler.py [hunk 7]
```diff
@@ -15,11 +16,17 @@ def schedule_events(event_list):
...
+
 def print_schedule(schedule):
     for hour, events in sorted(schedule.items()):
         print(f"{hour}:00 - {', '.join(events)}")
 
-events = [Event("Python Workshop", 19), Event("Data Science Meetup", 14), Event("Web Dev Seminar", 18)]
+
+events = [
+    Event("Python Workshop", 19),
+    Event("This should not be happening", 30),
```
Comment (dancer-yuma) [inline 04_error_handling_and_debugging/02_challenge/event_scheduler.py:27]: I added one line that represents an edge case.

**Review by dancer-yuma — COMMENTED** [review]
Self review

File: 02_classes_modules_and_packages/01_challenge/README.md (hunk 3 above)
Comment (abolinsky) [inline 02_classes_modules_and_packages/01_challenge/README.md:15]:
```suggestion
    - `retrieve_books()`: Returns a list of dictionaries that represents books in the library.
```

**Review by abolinsky — CHANGES_REQUESTED** [review]
Just that one small fix

**Review by abolinsky — APPROVED** [review]

**Pull request merged**